### PR TITLE
chore: bump alloy 4e22b9e

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/paradigmxyz/reth/issues/7702

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
great, fewer alloy changes is good

the most significant change was `to: Option<TxKind>` on the tx request.

this nuance has no impact on the server side and we can always use `unwrap_or_default`

**Comment by onbjerg** [comment]
@fgimenez #7885 is being merged which will have a small impact on this PR

**Comment by fgimenez** [comment]
> the most significant change was `to: Option<TxKind>` on the tx request.
> 
> this nuance has no impact on the server side and we can always use `unwrap_or_default`

yep, there were also other minor changes in alloy's serde. 

wrt to the new `to` type in `TransactionRequest`, we previously had `Option<Address>` and I assume that `None` here was interpreted as a create transaction. Now we have `Option<TxKind>`, we have `Some(TxKind::Call(to))` and `Some(TxKind::Create)`, how should we interpret `None` now? In the changes in this PR `None` means create too (same as `Some(TxCall::Create)`), let me know if that's correct.

File: crates/rpc/rpc/src/eth/api/transactions.rs [hunk 1]
```diff
@@ -901,6 +901,10 @@ where
             ..
         } = request;
 
+        let kind = match to {
+            Some(tx_kind) => tx_kind,
+            None => RpcTransactionKind::Create,
+        };
```
Comment (onbjerg) [inline crates/rpc/rpc/src/eth/api/transactions.rs:907]: i think `RpcTransactionKind` is the same as `alloy_primitives::TxKind` so we should be able to just do `to.unwrap_or(TxKind::Create)`?

**Review by onbjerg — APPROVED** [review]
looks sound to me defaulting to create, my nits can be looked at in a follow up

for clarity `None` means that the user did not specify what they wanted to do with the transaction request and the node should instead infer, the only thing we can probably infer is that it should be a create

File: crates/rpc/rpc-types-compat/src/transaction/mod.rs [hunk 2]
```diff
@@ -181,7 +181,7 @@ pub fn from_primitive_access_list(
 /// Convert [TransactionSignedEcRecovered] to [TransactionRequest]
 pub fn transaction_to_call_request(tx: TransactionSignedEcRecovered) -> TransactionRequest {
     let from = tx.signer();
-    let to = tx.transaction.to();
+    let to: Option<PrimitiveTransactionKind> = Some(tx.transaction.to().into());
```
[the remark this answers is not in the document]
Comment (fgimenez) [reply crates/rpc/rpc-types-compat/src/transaction/mod.rs]: yep, I was using the alias in this module, in any case not needed the type annotation, removed.

File: crates/rpc/rpc/src/eth/api/transactions.rs (hunk 1 above)
Comment (fgimenez) [reply crates/rpc/rpc/src/eth/api/transactions.rs]: good point, done

**Pull request merged**